raiden/network/rpc/client.py

```diff
@@ -509,18 +509,18 @@ def __init__(
 
         result = jsonrpc_client.call(
             'eth_getCode',
-            address_encoder(token_address),
+            token_address,
```

Comment:
At this point `token_address` should be a binary address. This is why the `address_encoder()` call is there. To turn it from binary into hex and send it off to the ethereum client via the RPC call.

You can't remove the `address_encoder()` from here. You are probably using something else in a wrong way in a higher part of the caling pipeline, or something else needs adjusting.

The general rule is:

Whatever is inside the raiden code, should be in binary. Then each module, like say the console, should know what kind of format to expect the addresses in and turn them to the common format which is binary. Same for when sending off data outside of raiden.